src/Orleans/Utils/WrappedException.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime.Internal
+{
+    /// <summary>Exception used to communicate with the storage provider, so that it throws this exception to its caller.</summary>
+    [Serializable]
+    public class WrappedException : OrleansException
```

Comment:
Internal/public is one dimension, but my main point is that, imo, this should be considered part of the memory storage provider implementation.  As it's an internal signal between the grain and the storage provider, it's not a proper 'Orleans exception', so should not inherit from that base class, and should not be in Orleans utils, but instead, packaged with the memory storage provider.